### DISCUSSION TITLE
Login com senha na área do cliente

### DISCUSSION THREAD

**Discussion by WillianMartinezS** [RamonSilva20/mapos#1693] [Ideas]
Olá galera.
Estou testando o MapOS faz um tempo e estou gostando muito do sistema. Aproveito para deixar meu elogio para nova interface, ficou sensacional, principalmente para uso através do celular.
Subi o MapOS para teste na hospedagem que utilizo e achei muito interessante a área do cliente. O problema em abrir o acesso público do MapOS através de uma hospedagem, é o fato de que qualquer pessoa pode entrar na área do cliente, cadastrar dados aleatórios sem passar por uma análise ou mesmo sem ficar em uma fila para que eu possa validar se os dados estão corretos e são reais, antes de efetivar o cadastro no meu banco de dados. 
Além disso, se por exemplo eu tiver um cliente cadastrado na minha base e os dados dele for fácil de descobrir através de uma pesquisa rápida pela internet, qualquer pessoa mal intencionada pode buscar um CNPJ e e-mail cadastrado na receita federal e com isso ter acesso as ordens de serviço **que podem conter informações sensíveis.**
Por isso acredito que seria muito interessante que o acesso a área do cliente fosse disponibilizada através de login com senha. Pensando também na segurança do sistema quanto ao acesso indevido ao banco de dados, acredito que a senha deveria ser criada a partir do administrador, por exemplo, eu sendo empresa prestadora de serviços faria o cadastro do meu cliente, gero um login e senha e passo essas informações para o cliente, a partir desse momento o cliente poderá alterar essa senha por uma de sua preferência (desde que atenda um requisito mínimo de complexidade), e assim passará a ter acesso a qualquer momento ao MapOS para consultar suas ordens de serviço, cobranças e atualizar dados cadastrais.

Agradeço a atenção e espero que meu feedback possa ajudar a melhorar a segurança desse sistema.

**Comment by WillianMartinezS** [comment]
Exemplo de acesso indevido ao cadastro de um cliente através de uma rápida pesquisa por seu nome através da internet, onde localizei o CNPJ e e-mail em um site qualquer.
![image](https://user-images.githubusercontent.com/101023951/156902825-a1dc3087-1454-498f-b58b-d07a5e114a16.png)

**Comment by willph** [comment]
Fiz uma meta no grupo do WhatsApp, caso bata a meta em doações(R$) para o projeto, estarei enviando essa alteração.

**Comment by willph** [comment]
Meta atingida (R$) em breve estarei disponibilizando.

**Comment by WillianMartinezS** [comment]
Olá, agradeço pelo retorno.
Sei que já bateram a meta, mas ainda assim posso dar alguma ajuda ($$)?

Esse grupo que vocês mencionam do WhatsApp, seria interno entre vocês devs ou é aberto para quem acompanha o projeto MapOS?

**Reply by willph** [reply]
Não. 

Liberado para quem quer tirar dúvida, contribuir com código, e tudo mais.

https://chat.whatsapp.com/L8iJYyfHbHbIkh13XMGtjV

**Reply by willph** [reply]
> Olá, agradeço pelo retorno. Sei que já bateram a meta, mas ainda assim posso dar alguma ajuda ($$)?
> 
> Esse grupo que vocês mencionam do WhatsApp, seria interno entre vocês devs ou é aberto para quem acompanha o projeto MapOS?

Sei que já bateram a meta, mas ainda assim posso dar alguma ajuda ($$)?
Respondendo sua pergunta é sim, tem via Vakinha, PIX, Catarse. O que você acha melhor pra você.

Já conseguiu testar senha?

Teve novas atualizações, e vai ter novas atualizações. 
Acompanhe o projeto, e não esqueça de deixar o sistema sempre atualizado.

**Reply by WillianMartinezS** [reply]
Em breve farei uma contribuição monetária.

Ainda não testei, vi no grupo que estavam modificando muitas coisas, fico feliz com o crescimento do projeto.
Em breve irei testar e darei feedback.

Muito obrigado por terem escolhido minha sugestão.

**Reply by dokti** [reply]
ainda é possível entrar neste grupo?

**Comment by willph** [comment]
Sim.

Mas o link foi modificado

Novo link.
https://chat.whatsapp.com/D571nPLqNq6JgIsmL6Gucl

Em qui, 20 de abr de 2023 22:00, dokti ***@***.***> escreveu:
